### PR TITLE
run tests in parallel mode

### PR DESCRIPTION
**Pull request by antip00**
Adding functionality to run tests in parallel.

### CODE REVIEW

File: tests/e2e/test_operator.py [hunk 1]
```diff
@@ -412,124 +505,206 @@ def check_remote_servers(self, chi, shards, trigger_event, cluster=""):
...
-@TestScenario
-@Name("test_008_1. Test operator restart")
-@Requirements(RQ_SRS_026_ClickHouseOperator_Managing_RestartingOperator("1.0"))
-def test_008_1(self):
-    with Check("Test simple chi for operator restart"):
-        test_operator_restart(
-            manifest="manifests/chi/test-008-operator-restart-1.yaml",
-            service="clickhouse-test-008-1",
-        )
+# @TestScenario
```
Comment (Slach) [inline tests/e2e/test_operator.py:508]: uncomment this test

File: tests/e2e/test_operator.py [hunk 2]
```diff
@@ -1952,13 +2296,39 @@ def test_020(self, step=1):
...
+            )
+
     test_020(step=1)
 
 
 @TestScenario
 @Name("test_020_2. Test multi-volume configuration, step=2")
 @Requirements(RQ_SRS_026_ClickHouseOperator_Deployments_MultipleStorageVolumes("1.0"))
 def test_020_2(self):
+    with Given("I create shell"):
```
Comment (Slach) [inline tests/e2e/test_operator.py:2319]: too much copy pasted code

Comment (antip00) [reply tests/e2e/test_operator.py]: Fixed.

File: tests/e2e/test_operator.py (hunk 1 above)
Comment (antip00) [reply tests/e2e/test_operator.py]: Fixed.

**Comment by alex-zaitsev** [comment]
Do not merge to 0.21.

**Pull request closed without merging**